header/header.go

```diff
@@ -223,7 +229,12 @@ func (eh *ExtendedHeader) UnmarshalBinary(data []byte) error {
 		return err
 	}
 
-	*eh = *out
+	*eh = ExtendedHeader{
+		RawHeader:    out.RawHeader,
+		Commit:       out.Commit,
+		ValidatorSet: out.ValidatorSet,
+		DAH:          out.DAH,
+	}
```

Comment:
Isn't this an equivalent of what it was before, but without a method?

Reply:
(not relevant)

Reply:
Yes, it's. We did shallow copy before.